test/js/data/fill_bucket.test.js

```diff
@@ -52,3 +52,74 @@ test('FillBucket', function(t) {
 
     t.end();
 });
+
+test('FillBucket - feature split across array groups', function (t) {
+    var layer = new StyleLayer({
+        id: 'test',
+        type: 'fill',
+        layout: {},
+        paint: { 'fill-color': {
+            stops: [[0, 'red'], [1, 'blue']],
+            property: 'foo'
+        } }
+    });
+
+    // this, plus the style function, sets things up so that
+    // populatePaintArrays iterates through each vertex
+    layer.updatePaintTransition('fill-color', [], {});
+
+    var bucket = new FillBucket({
+        buffers: {},
+        layer: layer,
+        childLayers: [layer]
+    });
+    bucket.createArrays();
+
+    // add an initial feature
+    bucket.addFeature(createBigFeature(32768));
+    // now add a feature that will break across the group boundary
+    bucket.addFeature(createBigFeature(32768 + 1));
+
+    // check that every vertex's color values match the first vertex
+    var groups = bucket.arrayGroups.fill;
+    var expected = groups[0].paint.test.get(0);
+    expected = [
+        expected['a_color0'],
+        expected['a_color1'],
+        expected['a_color2'],
+        expected['a_color3']
+    ];
+
+    for (var g = 0; g < groups.length; g++) {
+        var group = groups[g];
+        for (var i = 0; i < group.paint.test.length; i++) {
+            var vertex = group.paint.test.get(i);
+            var color = [
+                vertex['a_color0'],
+                vertex['a_color1'],
+                vertex['a_color2'],
+                vertex['a_color3']
+            ];
+
+            // do this instead of t.deepEqual so as not to pollute test output
+            // with > 65536 assertions
+            if (expected.join(',') !== color.join(',')) {
+                t.fail('Vertex ' + i + ' does not match first vertex; found ' + color + ', but expected ' + expected);
+                t.end();
+                return;
+            }
+        }
+    }
+
+    t.end();
+
+
+});
```

Comment:
👍 Yeah, agreed, we should really just stick to testing around the boundaries.    I'd add a fourth one:
 - test _last_ vertex (and I'll change the sizes so that this is very clearly distinct from vertex 65535, which we'll be testing because it's the first vertex of the new group)